resources/Microsoft.Windows.Assertion/Microsoft.Windows.Assertion.psm1

```diff
@@ -0,0 +1,461 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+enum PnPDeviceState
+{
+    OK
+    ERROR
+    DEGRADED
+    UNKNOWN
+}
+
+[DSCResource()]
+class OsEditionId
+{
+    # Key required. Do not set.
+    [DscProperty(Key)]
+    [string]$SID
```

Comment:
I would change the mandatory property to become the Key. We don't need an SID anymore. I think this can be applied to all the new DSC resources below.

Reply:
Will do!